src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocTypeCheck.java

```diff
@@ -0,0 +1,213 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FileContents;
+import com.puppycrawl.tools.checkstyle.api.Scope;
+import com.puppycrawl.tools.checkstyle.api.TextBlock;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.AnnotationUtil;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+import com.puppycrawl.tools.checkstyle.utils.ScopeUtil;
+
+/**
+ * <p>
+ * Checks for missing Javadoc comments for class, enum, interface, and annotation interface
+ * definitions. The scope to verify is specified using the {@code Scope} class and defaults
+ * to {@code Scope.PUBLIC}. To verify another scope, set property scope to one of the
+ * {@code Scope} constants.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code scope} - specify the visibility scope where Javadoc comments are checked.
+ * Default value is {@code public}.
+ * </li>
+ * <li>
+ * Property {@code excludeScope} - specify the visibility scope where Javadoc comments are not
+ * checked. Default value is {@code null}.
+ * </li>
+ * <li>
+ * Property {@code allowedAnnotations} - specify the list of annotations that allow missed
+ * documentation. Only short names are allowed, e.g.{@code Generated}. Default value is
+ * {@code Generated}.
+ * </li>
+ * <li>
+ * Property {@code tokens} - tokens to check Default value is:
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#INTERFACE_DEF">INTERFACE_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CLASS_DEF">CLASS_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_DEF">ENUM_DEF</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ANNOTATION_DEF">ANNOTATION_DEF</a>.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the default check:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingJavadocType"/&gt;
+ * </pre>
+ * <p>
+ * To configure the check for {@code private} scope:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingJavadocType"&gt;
+ *   &lt;property name="scope" value="private"/&gt;
```

Comment:
lets have real life example.
Can you imagine some rule that demand javadoc on private classes and not demand javadoc on other scopes.

Reply:
See previous comment. Private doesn't mean only private.
Since this does everything, I don't think this example has to change, If anything documentation on `property_types.html` should be expanded since other modules can use this type.